### PR TITLE
Wrap evaluate and push_term!/finish

### PR DESCRIPTION
**Pull request by thofma**

### CODE REVIEW

File: src/flint/fmpq_mpoly.jl [hunk 1]
```diff
@@ -789,6 +789,37 @@ f values")
...
+   length(bs) != nvars(R) &&
+      error("Number of variables does not match number of values")
+
+   c = S()
+   fl = ccall((:fmpq_mpoly_compose_fmpq_poly, libflint), Cint,
+              (Ref{fmpq_poly}, Ref{fmpq_mpoly}, Ptr{Ref{fmpq_poly}},
+               Ref{FmpqMPolyRing}), c, a, bs, R)
+   fl == 0 && error("Something wrong in evaluation.")
+   return c
+end
```
Comment (tthsqe12) [inline src/flint/fmpq_mpoly.jl:821]: This whole thing seems to be a duplicate. ditto for ZZ

Comment (thofma) [reply src/flint/fmpq_mpoly.jl:821]: Duplicate of what? I did not get an "method overwritten" error.

Comment (tthsqe12) [reply src/flint/fmpq_mpoly.jl:821]: My bad

**Comment by thofma** [comment]
Good to go @tthsqe12?

**Review by tthsqe12 — APPROVED** [review]
Looks good. However, eventually we should be able to remove these evaluation methods because they can be done better generically in AA.

**Pull request merged**